### PR TITLE
Asset Browser Tests

### PR DESCRIPTION
**Pull request by AMZN-Igarri**
Added some tests for the Asset Browser which currently had none.
image

### CODE REVIEW

File: Code/Framework/AzToolsFramework/Tests/UI/AssetBrowserTests.cpp [hunk 1]
```diff
@@ -0,0 +1,392 @@
...
+        CreateProduct(m_productIDs.at(0), sourceUuid_1, "Product_1_0");
+        CreateProduct(m_productIDs.at(1), sourceUuid_1, "Product_1_1");
+
+        QString resultMessage = QString();
+        AZStd::function<void(const QString&)> printer = [&resultMessage](const QString& message)
+        {
+            resultMessage += message + "\n";
+        };
+        PrintModel(m_assetBrowserComponent->GetAssetBrowserModel(), printer);
+        [[maybe_unused]] AZStd::vector<QString> createdHierarchySections = GetVectorFromFormattedString(resultMessage);
```
Comment (hultonha) [inline Code/Framework/AzToolsFramework/Tests/UI/AssetBrowserTests.cpp:247]: question: I take it the plan is to do something useful with this in a follow-up PR?

File: Code/Framework/AzToolsFramework/Tests/UI/AssetBrowserTests.cpp [hunk 2]
```diff
@@ -0,0 +1,392 @@
...
+
+        product.second.m_subID = aznumeric_cast<AZ::u32>(productID);
+        product.second.m_productName = productName;
+
+        GetRootEntry()->AddProduct(product);
+    }
+
+    void AssetBrowserTest::SetupAssetBrowser()
+    {
+
```
Comment (hultonha) [inline Code/Framework/AzToolsFramework/Tests/UI/AssetBrowserTests.cpp:181]: uber nit: Please remove this blank line (might be worth running clang-format over this file too)

**Review by hultonha — APPROVED** [review]
I think this looks like a fantastic start and it's going to be great to expand on these tests in future, nice work getting these changes to this stage 🙂

**Review by jonawals — APPROVED** [review]
Great baseline to build up from with future tests 👍

File: Code/Framework/AzToolsFramework/Tests/UI/AssetBrowserTests.cpp (hunk 1 above)
Comment (AMZN-Igarri) [reply Code/Framework/AzToolsFramework/Tests/UI/AssetBrowserTests.cpp:247]: Yes! That's the plan

**Pull request merged**